src/prototype_room_mineral.js

```diff
@@ -221,3 +221,87 @@ Room.prototype.handleTerminal = function() {
   delete this.memory.mineralOrder[roomOtherName];
   return true;
 };
+
+Room.prototype.cleanupNotEnoughResources = function() {
+  if (!this.memory.reaction) {
+    return false;
+  }
+  const lab1 = Game.getObjectById(this.memory.reaction.labs[1]);
+  const lab2 = Game.getObjectById(this.memory.reaction.labs[2]);
+  const mineralCreep = this.findMyCreepsOfRole('mineral').length ? this.findMyCreepsOfRole('mineral')[0] : false;
```

Comment:
```suggestion
  const mineralCreeps = this.findMyCreepsOfRole('mineral');
  if (mineralCreeps.length === 0) {
      return;
  }
  const mineralCreep = mineralCreeps[0];
```

(minimize find calls, to save CPU)